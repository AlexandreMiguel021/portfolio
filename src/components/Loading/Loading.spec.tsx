import { render } from '@testing-library/react'
import { Loading } from '.'
import '@testing-library/jest-dom'

describe('Loading', () => {
	it('should render a Loading correctly', () => {
		const { container } = render(<Loading loading={false} />)
		expect(container).toMatchSnapshot()
	})

	it('should render a loading screen if loading is true', () => {
		const { container } = render(<Loading loading={true} />)
		expect(container.querySelector('.css-a4owti')).toBeInTheDocument()
	})

	it('should not render a loading screen if loading is false', () => {
		const { container } = render(<Loading loading={false} />)
		expect(container.querySelector('.css-a4owti')).not.toBeInTheDocument()
	})
})
