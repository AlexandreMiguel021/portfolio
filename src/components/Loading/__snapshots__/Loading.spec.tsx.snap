// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render a Loading correctly 1`] = `
<div>
  <div
    class="loading"
  >
    <div
      class="sweet-loading"
    />
  </div>
</div>
`;
